### PR TITLE
add aop autotest-plan-before

### PR DESCRIPTION
**Pull request by ZLValien**
#### What type of this PR

/kind feature

#### What this PR does / why we need it:

when autotest-plan is executed,report configuration

### CODE REVIEW

File: modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		Meta:       meta,
+	})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (p *provider) Init(ctx servicehub.Context) error {
+	p.Bundle = bundle.New(bundle.WithDOP())
```
Comment (sfwn) [inline modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go:93]: pipeline is not allowed to invoke dop.

**Review by sfwn — CHANGES_REQUESTED** [review]

File: modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go (hunk 1 above)
Comment (sfwn) [reply modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go:93]: use event to decouple.

File: modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+	autoTestGlobalConfigListRequest.Scope = "project-autotest-testcase"
+	autoTestGlobalConfigListRequest.UserID = ctx.SDK.Pipeline.PipelineExtra.Snapshot.PlatformSecrets["dice.user.id"]
+	configs, err := p.Bundle.ListAutoTestGlobalConfig(autoTestGlobalConfigListRequest)
+	if err != nil {
+		return err
+	}
+
+	meta := make(apistructs.PipelineReportMeta)
+	for _, v := range configs {
+		if v.Ns == ctx.SDK.Pipeline.PipelineExtra.Extra.ConfigManageNamespaces[0] {
```
[the remark this answers is not in the document]
Comment (ZLValien) [reply modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go:71]: configManageNamespaces":["autotest^scope-project-autotest-testcase^scopeid-x^xxxxxxx"]
Now it has only one

File: modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/modules/dop/services/autotest"
+	"github.com/erda-project/erda/modules/pipeline/aop"
+	"github.com/erda-project/erda/modules/pipeline/aop/aoptypes"
+)
+
+// +provider
+type provider struct {
+	aoptypes.PipelineBaseTunePoint
+	CmsService pb.CmsServiceServer `autowired:"erda.core.pipeline.cms.CmsService"`
```
Comment (kakj-go) [inline modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go:33]: not used

File: modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+func (p *provider) Name() string { return "testplan-before" }
+
+func (p *provider) Handle(ctx *aoptypes.TuneContext) error {
+	// source = autotest
+	if ctx.SDK.Pipeline.PipelineSource != apistructs.PipelineSourceAutoTest || ctx.SDK.Pipeline.IsSnippet {
+		return nil
+	}
+
+	// filter PipelineYmlName is not autotest-plan-xxx
+	testPlanID, err := checkPipelineYmlName(ctx.SDK.Pipeline.PipelineYmlName)
```
Comment (kakj-go) [inline modules/pipeline/aop/plugins/pipeline/testplan_before/plugin.go:45]: hava apistructs.PipelineSourceAutoTestPlan prefix just enough

**Review by sfwn — APPROVED** [review]

**Pull request merged**